### PR TITLE
validate location from fwrj

### PR DESCRIPTION
**Pull request by ticoann**
just prevent situation happened using T0_CERN_CH

### CODE REVIEW

**Comment by ticoann** [comment]
@amaltaro, Alan, please review

File: src/python/WMComponent/JobAccountant/AccountantWorker.py [hunk 1]
```diff
@@ -794,7 +795,13 @@ def handleWMBSFiles(self, wmbsFilesToBuild, parentageBinds):
                 runLumiBinds.append({'lfn': lfn, 'runs': wmbsFile['runs']})
 
             if len(wmbsFile.getLocations()) > 0:
-                fileLocations.append({'lfn': lfn, 'location': wmbsFile.getLocations()[0]})
+                outpnn = wmbsFile.getLocations()[0]
+                if self.pnn_to_psn.get(outpnn, None):
```
Comment (amaltaro) [inline src/python/WMComponent/JobAccountant/AccountantWorker.py]: None is redundant here.

**Comment by amaltaro** [comment]
Code looks Ok, I'm just not sure whether we should really break the component if this case happens again. I'd instead print a warning and fail the job (possible adding a new error/message in WMExceptions).

**Comment by ticoann** [comment]
I think we should crash the component. If job runs somewhere we don't have the record of, 
1. it is completely screwed up
2. Or we don't have the proper site record in our table.
I think it would be mostly 2 case (although proper way is to have SSB record), then we need add the site in the table manually and restart the component.

If we just print error and move on, we need to do manual surgery when case 2 cause the problem.
I think it actually shouldn't happen (at least very often). But it happens it might be easier to fix by crashing Component

**Comment by ticoann** [comment]
I will update the patch to fix unit tests.

File: src/python/WMComponent/JobAccountant/AccountantWorker.py (hunk 1 above)
Comment (ticoann) [inline src/python/WMComponent/JobAccountant/AccountantWorker.py]: Yes, I just make consistent with other code in the module. :smile:

**Comment by amaltaro** [comment]
> If we just print error and move on, we need to do manual surgery when case 2 cause the problem.

we do not move on. We permanently fail this job, report failures to wmstats and someone  fix either a) the assignment or some black magic script that was screwing with jobs or b) fix resource control/sitedb.

I'm just trying to get rid of ALL the unneeded component crashes.
On the other hand, I agree that this should happen not more than once/twice per year, so it's ok'ish to crash the component if you prefer (at least we would be able to make a hot fix and not waste cpu time)

**Comment by ticoann** [comment]
> e do not move on. We permanently fail this job, report failures to wmstats and someone fix either a) the assignment or some black magic script that was screwing with jobs or b) fix resource control/sitedb.

I see, I think a) case it makes sense to failed the jobs  but b) case, it maybe better to fix resource and not redo the whole processing.  I am guessing most case will be b). but I might be wrong.

**Comment by amaltaro** [comment]
Patch looks good, though I'm running a couple quick workflows just to make sure nothing breaks.
If it's Ok, I think this one deserves to be backported to 1.0.14

**Comment by amaltaro** [comment]
@ticoann feel free to merge this one

**Comment by ticoann** [comment]
test this please

**Pull request merged**